File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = ''
    
    numbers = {
        'zero': '0',
        'one': '1',
        'two': '2',
        'three': '3',
        'four': '4',
        'five': '5',
        'six': '6',
        'seven': '7',
        'eight': '8',
        'nine': '9'
    }
    text = ''
    for i in s:
        if text in numbers:
            answer += numbers[text]
            text = ''
        
        # 숫자 x
        if i.isdecimal() == False: 
            text += i
        
        # 숫자
        else :
            answer += i
    
    # 마지막 확인
    if text in numbers:
        answer += numbers[text]
     
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        
        String[] nums = {"zero","one","two","three","four","five","six","seven","eight","nine"};
        
        for (int i = 0; i < 10; i++) {
            s = s.replace(nums[i], Integer.toString(i));
        }
 
        return Integer.parseInt(s);  // int로 변환하여 return
    }
}